src/clients/python/experimental_api_v2/library/httpclient.py

```diff
@@ -107,198 +104,447 @@ def close(self):
 
+        Parameters
+        ----------
+        headers: dict
+            Optional dictionary specifying the headers that must be
+            sent alongwith the request.
+
         Returns
         -------
-        object
-            The JSON object holding the metadata
+        dict
+            The JSON dict holding the metadata.
 
         Raises
         ------
-        InferenceServerException
+        Exception
             If unable to get server metadata.
 
         """
+        response = self._client_stub.get("v2", headers=headers)
+        raise_if_error(response)
+        metadata = json.loads(response.read())
+        return metadata
 
-        response = self._client_stub.get("/v2")
-        self._last_request_id = raise_if_error(response['NV-Status'])
-
-        status = json.loads(self.response.read())
-        return status
-
-    def get_model_metadata(self, model_name, model_version=-1):
+    def get_model_metadata(self, model_name, model_version="", headers={}):
         """Contact the inference server and get the metadata for specified model.
 
         Parameters
         ----------
         model_name: str
             The name of the model
+        model_version: str
+            The version of the model to get metadata. The default value
+            is an empty string which means then the server will choose
+            a version based on the model and internal policy.
+        headers: dict
+            Optional dictionary specifying the headers that must be
+            sent alongwith the request
 
         Returns
         -------
-        object
-            The JSON object holding the model metadata
+        dict
+            The JSON dict holding the metadata.
 
         Raises
         ------
-        InferenceServerException
+        Exception
             If unable to get model metadata.
 
         """
-        #response = self._client_stub.get("/api/status/" + model_name +
-        #                                      "?format=json")
-        #self._last_request_id = raise_if_error(response['NV-Status'])
-
-        #status = json.loads(response.read())
-        #return status
-        raise_error('Not implemented')
-        return None
-
-    def load_model(self, model_name):
-        """Request the inference server to load or reload specified model.
+        if not model_version:
+            request_uri = "v2/models/{}".format(model_name)
+        else:
+            request_uri = "v2/models/{}/versions/{}".format(
+                model_name, model_version)
+        response = self._client_stub.get(request_uri, headers=headers)
+        raise_if_error(response)
+        metadata = json.loads(response.read())
+        return metadata
+
+    def infer(self,
+              inputs,
+              model_name,
+              model_version="",
+              outputs=None,
+              request_id=None,
+              parameters=None,
+              headers={}):
```

Comment:
Using a mutable type to default `headers` is dangerous.  

See: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

(It basically behaves like a static function variable in C++)